### PR TITLE
switch affix_item_type bit flag enum to class format

### PR DESCRIPTION
**Pull request by DakkJaniels**
Same as that spellflag enum, switched over the affix_item_type to the same format.

### CODE REVIEW

File: Source/items.cpp [hunk 1]
```diff
@@ -1201,7 +1201,7 @@ void GetItemPower(Item &item, int minlvl, int maxlvl, affix_item_type flgs, bool
 				continue;
 			if (onlygood && !ItemPrefixes[j].PLOk)
 				continue;
-			if (flgs == PLT_STAFF && ItemPrefixes[j].power.type == IPL_CHARGES)
+			if (HasAnyOf(flgs,AffixItemType::Staff) && ItemPrefixes[j].power.type == IPL_CHARGES)
```
Comment (AJenbo) [inline Source/items.cpp:1204]:
```suggestion
			if (HasAnyOf(flgs, AffixItemType::Staff) && ItemPrefixes[j].power.type == IPL_CHARGES)
```

Comment (DakkJaniels) [reply Source/items.cpp]: yup just noticed it. I wish Visual Studio would fix that before it uploads the commit...

File: Source/itemdat.cpp [hunk 2]
```diff
@@ -226,94 +226,94 @@ const PLStruct ItemPrefixes[] = {
 	// clang-format off
 	// PLName,              { type,                 param1, param2 }, PLMinLvl, PLIType,                                                         PLGOE,    PLDouble, PLOk,  minVal, maxVal, multVal
```
Comment (AJenbo) [inline Source/itemdat.cpp:227]: Probably the header needs some adjustment now?

Comment (DakkJaniels) [reply Source/itemdat.cpp]: missed that - fixed

**Review by AJenbo — APPROVED** [review]

**Pull request merged**